core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -711,6 +711,15 @@ public NativeImageInvokerInfo build() {
                     }
                 }
 
+                // only available in GraalVM 22.3.0 and better.
+                if (graalVMVersion.compareTo(GraalVM.Version.VERSION_22_3_0) >= 0) {
+                    // For build time information
+                    nativeImageArgs.add("-H:+CollectImageBuildStatistics");
+                    nativeImageArgs.add("-H:ImageBuildStatisticsFile=" + nativeImageName + "-timing-stats.json");
+                    // For getting the build output stats as a JSON file
+                    nativeImageArgs.add("-H:BuildOutputJSONFile=" + nativeImageName + "-build-output-stats.json");
```

Comment:
Hmmm, I have a last minute question though, where are these files generated? In `target/` together with the image, right, not at the root?

Reply:
Yes, in the `target` folder:

```
$ find target/ -name \*.json
target/rest-json-quickstart-1.0.0-SNAPSHOT-native-image-source-jar/rest-json-quickstart-1.0.0-SNAPSHOT-runner-timing-stats.json
target/rest-json-quickstart-1.0.0-SNAPSHOT-native-image-source-jar/rest-json-quickstart-1.0.0-SNAPSHOT-runner-build-output-stats.json
```

Reply:
Also please let me know if you need me to rebase before merging.